polkadot/node/core/pvf/src/error.rs

```diff
@@ -27,35 +27,42 @@ pub enum ValidationError {
 	/// pre-checking enabled only valid runtimes should ever get enacted, so we can be
 	/// reasonably sure that this is some local problem on the current node. However, as this
 	/// particular error *seems* to indicate a deterministic error, we raise a warning.
+	#[error(transparent)]
 	Preparation(PrepareError),
 	/// The error was raised because the candidate is invalid. Should vote against.
-	Invalid(InvalidCandidate),
+	#[error(transparent)]
```

Comment:
Can we have specific error messages here instead of the fall-through `transparent`?

Reply:
Let me explain the motivation:)

1. `InternalValidationError` already has `validation: ` prefix in its messages
2. `ValidationError` has 1-1 match between enum variants and wrapped sub errors
3. As a novice to the codebase, I am not sure how much freedom I have over the error messages

So I decided to stick with the safest solution in my opinion. But of course, if you think we should expand messages on wrapper, I will do :)

Reply:
Good points! If you want though, you can change this:

- For preparation, add a message like `preparation error for validation: {0}`
- For internal error, maybe change the message to `internal validation error: {0}`

> As a novice to the codebase, I am not sure how much freedom I have over the error messages

Appreciate the humility and respect for the codebase. :)

Reply:
Also not sure if the change is necessary TBH so I'll leave it to your judgment. I think it's fine to err on the side of being a bit too verbose: error messages are rare and the more info they contain, the better.

Reply:
Agree! I've [decided to use](https://github.com/paritytech/polkadot-sdk/pull/2958/commits/471a768493f2b3629bffb7ff59cc63cc963bf317) `candidate validation:` prefix for `ValidationError`. Thus it looks consistent